Improve behavior of namespace flags in operator-sdk run and operator-sdk test local subcommands (#2617)
operator-framework/operator-sdk: doc/cli/operator-sdk_cleanup.md

### Options

```
      --kubeconfig string           The file path to kubernetes configuration file. Defaults to location specified by $KUBECONFIG, or to default file rules if not set

>       --namespace string            The namespace where the operator watches for changes.

Reviewer: ```suggestion
      --namespace string            (Deprecated) - The namespace where the operator watches for changes.
```

Reviewer: Did you not change this as well?

Reviewer: I think we need to change all to keep the same behaviour across the project. Otherwise is hard to know when use  1 or another. Is it make sense?

Author: done

      --olm                         The operator to be cleaned up is managed by OLM in a cluster. Cannot be set with another cleanup-type flag (default true)

>       --operator-namespace string   [olm only] The namespace where operator resources are created in --olm mode. It must already exist in the cluster or be defined in a manifest passed to IncludePaths.

Reviewer: same ^

>       --operator-namespace string   [olm only] The namespace where operator resources are created in --olm mode. It must already exist in the cluster or be defined in a manifest passed to IncludePaths.

Reviewer: ```suggestion
      --operator-namespace string   [olm only] The namespace where the operator is deployed in. 
```
